clippy_lints/src/lib.rs

```diff
@@ -491,31 +390,102 @@ pub fn read_conf(sess: &Session) -> Conf {
     conf
 }
 
+pub(crate) enum LintCategory {
+    Cargo,
+    Complexity,
+    Correctness,
+    Nursery,
+    Pedantic,
+    Perf,
+    Restriction,
+    Style,
+    Suspicious,
+    #[cfg(feature = "internal")]
+    Internal,
+}
+
+pub(crate) struct LintInfo {
+    /// Double reference to maintain pointer equality
+    lint: &'static &'static Lint,
```

Comment:
Figuring this out was quite an ordeal, I'm not exactly sure what was going on with just `&'static Lint`

`static`s seem to deeply(?) copy other `static`s if they're in a separate module, even for non-`Copy` structs

```rust
// m.rs
pub static ONE: &crate::S = &crate::S {
    str: "number one",
};
```

```rust
// main.rs
mod m;

use m::ONE;

pub struct S {
    str: &'static str,
}

static TWO: &S = ONE;

fn main() {
    println!("{:p}", ONE);
    println!("{:p}", TWO);

    println!();

    // Even the str contents are copied
    println!("{:p}", ONE.str.as_ptr());
    println!("{:p}", TWO.str.as_ptr());
}
```

prints

```
0xaaaad4f50558
0xaaaad4f50510

0xaaaad4f45c3b
0xaaaad4f45c30
```

Reply:
I guess the answer is fairly simple actually, `SOME_STATIC` acts the same as a const, only `&SOME_STATIC` has a stable address